### DISCUSSION TITLE
IPv6 not blocked

### DISCUSSION THREAD

**Discussion by henrixd7** [evilsocket/opensnitch#1169] [General]
I have hosts-file type blocklist with deny rule 000-block-domains as instructions told to make.
Blocklist has this line:
`0.0.0.0 cdn.cookielaw.org`

With this only udp gets blocked but udp6 and tcp6 is not, allowing the connection go through.
![Screenshot_20240814_043953](https://github.com/user-attachments/assets/1f81ec9f-e996-4c6f-8ac8-0941d3374b03)

How should I configure this to get domain blocked entirely?

**Comment by henrixd7** [comment]
Either I'm doing something really wrong or the program is not working quite right. I copied the blocklist rule and blocked it separately for udp, udp6, tcp6. Nevertheless, the traffic can still pass.

000-block-domains only matches udp requests, also blocking tcp from connecting, but udp6 and tcp6 match the permissive rule. Is that possibly a matter of prioritizing these rules? The 000-block-domains rules are listed first with id numbers 1, 2, 3 and 4.

I might be an idiot, but I don't understand.

![Screenshot_20240820_041208](https://github.com/user-attachments/assets/13f596c6-813e-42c3-8255-3ecd6120a641)
![Screenshot_20240820_041311](https://github.com/user-attachments/assets/48e00c8e-f877-4a22-a844-24d49490367a)

**Comment by gustavo-iniguez-goya** [comment]
000-block-domains should be a priority rule, Rules editor dialog -> [x] Priority rule.

If that change does still not work, edit the unbound rule, to allow connections from unbound only to port 53.

**Comment by henrixd7** [comment]
![image](https://github.com/user-attachments/assets/2b65d460-f02a-474a-9064-86e3570ef740)
Yes, it is a priority rule. It's now set to reject but same happens with deny too, only that it takes longer to timeout. I haven't been able to get it work with port restriction to unbound either.

So this debug log below shows that there's domain list match: `domain list match: www.livejasmin.com, /etc/blocklists/lists/steven-black.txt`. Then it first seems to block (udp) connection correctly with rule 000-block-domains: `✘ /app/lib/librewolf/librewolf -> 32867:127.0.0.1 => www.livejasmin.com (127.0.0.1):53, mark: 0 (000-block-domains)` but later despite matching the blocklist, also matches the allow rule allow-until-restart-simple-app-lib-librewolf-librewolf and lets (udp6) connection through: `✔ /app/lib/librewolf/librewolf -> 42871:::1 => www.livejasmin.com (::1):53, mark: 0 (allow-until-restart-simple-app-lib-librewolf-librewolf)`

> [2025-01-02 07:28:56]  DBG  domain list match: www.livejasmin.com, /etc/blocklists/lists/steven-black.txt
[2025-01-02 07:28:56]  DBG  Net list match: 127.0.0.1, 127.0.0.0/8
[2025-01-02 07:28:56]  DBG  ✘ /app/lib/librewolf/librewolf -> 32867:127.0.0.1 => www.livejasmin.com (127.0.0.
1):53, mark: 0 (000-block-domains)
[2025-01-02 07:28:56]  DBG  new connection udp6 => 42871:::1 -> ::1 (www.livejasmin.com):53 uid: 1000, mark: 
0
[2025-01-02 07:28:56]  DBG  [ebpf conn] not in cache, but in execEvents: udp642871::1::153, 4197 -> /app/lib/
librewolf/librewolf
[2025-01-02 07:28:56]  DBG  [ebpf conn] adding item to cache: udp642871::1::153
[2025-01-02 07:28:56]  DBG  domain list match: www.livejasmin.com, /etc/blocklists/lists/steven-black.txt
[2025-01-02 07:28:56]  DBG  ✔ /app/lib/librewolf/librewolf -> 42871:::1 => www.livejasmin.com (::1):53, mark:
 0 (allow-until-restart-simple-app-lib-librewolf-librewolf)
[2025-01-02 07:28:56]  DBG  New DNS record: 109.71.161.200 -> www.livejasmin.com

So for some reason udp6 doesn't seem to follow same priority rules than udp. I did upgrade from 1.6.6 to 1.6.7 but that didn't change anything. Having separate block-rules with defined protocol settings didn't work either.

There are some errors like this: `DBG  error deleting ebpf entry: unable to delete element: no such file or directory` in the log but I don't know if they are significant.

Here's full log from those requests: [opensnitch.log](https://github.com/user-attachments/files/18288811/opensnitch.log)

**Comment by TriMoon** [comment]
> I have hosts-file type blocklist with deny rule 000-block-domains as instructions told to make.
> Blocklist has this line:
> `0.0.0.0 cdn.cookielaw.org`

Can you provide the exact configuration settings you used?
Because IIRC, a file with hostnames per line does not use `0.0.0.0` as first part, OpenSnitch does not use a `/etc/hosts` type file...

Looking at https://github.com/evilsocket/opensnitch/wiki/block-lists#supported-list-types, it indeed is mentioned as being supported, nevertheless i suggest you to use only hostnames per line and try that..
If that works, create a new issue about the hosts-type file not working :wink:

**Reply by henrixd7** [reply]
I'm using [https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts](https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts) list which is in hosts-format. These lists do blocking by setting the IPs as 127.0.0.1 or 0.0.0.0 and thus prevent routing when they are used in /etc/hosts.conf.

Instructions in [https://github.com/evilsocket/opensnitch/wiki/block-lists#lists-of-domains](https://github.com/evilsocket/opensnitch/wiki/block-lists#lists-of-domains) states that format should be supported:
![image](https://github.com/user-attachments/assets/e17bb138-6be6-4b2f-baa3-b97a83398dd8)

These lists are used in 000-block-domains (priority) rule by configuring them in "List of domains/IPs"-tab on rule settings.
![image](https://github.com/user-attachments/assets/0850600c-799c-498a-89b8-041a85085037)
There's also list of network ranges which is in different format. However, I don't think that this part is the problem since domains are matched correctly to these lists and UDP is blocked as it should. Only UDP6 doesn't get blocked.

**Comment by henrixd7** [comment]
Oh, now it's working. I removed network ranges and only left the domain list configuration and now it blocks domains properly. So, it really was just a stupid user/configuration error after all. I really thought that I already tried that previously but I guess not. Deny and reject rules both work now. Cool.

**Comment by TriMoon** [comment]
If it was your own mistake, and it's fixed now, be so kind and close your issue :wink:
